### PR TITLE
Add cluster_info to the GET /_internal/desired-balance endpoint output

### PR DESCRIPTION
**Pull request by idegtiarenko**
This output would allow us to reconstruct RoutingAllocation locally in order to troubleshot unbalanced clusters

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/DesiredBalanceResponse.java [hunk 1]
```diff
@@ -35,19 +36,23 @@
 public class DesiredBalanceResponse extends ActionResponse implements ChunkedToXContentObject {
 
     private static final TransportVersion CLUSTER_BALANCE_STATS_VERSION = TransportVersion.V_8_7_0;
+    private static final TransportVersion CLUSTER_INFO_VERSION = TransportVersion.V_8_8_0;
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/DesiredBalanceResponse.java:39]: I wonder if it is worth also backporting to 8.7?

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/DesiredBalanceResponse.java:39]: It'd certainly be nice to have ASAP but it's not really a bugfix so I think we shouldn't backport.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/part-2

**Pull request merged**